nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -1166,37 +1468,83 @@ def get_query_type():
-            contained_1_in_2.output_result + ':' + contained_1_in_2.output_explanation
-        explanation_not_contained_other_self = \
-            contained_2_in_1.output_result + ':' + contained_2_in_1.output_explanation
         if contained_1_in_2.bool_result and contained_2_in_1.bool_result:
             return QueryAnswer(bool_result=True,
                                output_result=f'The two network configurations {self.name1} and {self.name2} '
                                              'are semantically equivalent.',
-                               numerical_result=3)
+                               numerical_result=3), None, None
+
+        sub_query_answer1 = QueryAnswer(output_result=contained_1_in_2.output_result,
+                                        output_explanation=contained_1_in_2.output_explanation)
+        sub_query_answer2 = QueryAnswer(output_result=contained_2_in_1.output_result,
+                                        output_explanation=contained_2_in_1.output_explanation)
         if not contained_1_in_2.bool_result and not contained_2_in_1.bool_result:
-            output_explanation = explanation_not_contained_self_other + '\n\n' + explanation_not_contained_other_self
             return QueryAnswer(bool_result=False,
                                output_result=f'Neither network configuration {self.name1} and {self.name2} '
-                                             'are contained in the other.\n',
-                               output_explanation=output_explanation, numerical_result=0)
+                                             f'are contained in the other', numerical_result=0), sub_query_answer1,\
+                sub_query_answer2
         if contained_1_in_2.bool_result:
             return QueryAnswer(bool_result=False,
-                               output_result=f'Network configuration {self.name1} is a proper subset of {self.name2}.\n',
-                               output_explanation=explanation_not_contained_other_self, numerical_result=2)
+                               output_result=f'Network configuration {self.name1} is a proper'
+                                             f' subset of {self.name2}', numerical_result=2), None, sub_query_answer2
         # (contained_2_in_1)
         return QueryAnswer(bool_result=False,
-                           output_result=f'Network configuration {self.name2} is a proper subset of {self.name1}.\n',
-                           output_explanation=explanation_not_contained_self_other, numerical_result=1)
+                           output_result=f'Network configuration {self.name2} is a proper subset of {self.name1}',
+                           numerical_result=1), sub_query_answer1, None
+
+    def execute_and_write_output_in_required_format(self, cmd_line_flag=False):
+        # TwoWayContainmentQuery executes two Containment sub queries
+        query_answer, sub_query_answer_1, sub_query_answer_2 = self.exec()
```

Comment:
cannot you use the numerical result from `query_answer `instead of having to use `sub_query_answer_1 `and `sub_query_answer_2 `?

Reply:
as mentioned above, here I use the `sub_query_answer_1` and `sub_query_answer_2` as objects for storing the `output_result` and `output_explanation` data of each `ContainmentQuery.exec()` Call.
as I need them for writing the correct output 
i.e. if both config1 is not contained in config2 and config2 is not contained in config1 , I need the separate explanations of each ContainmentQuery.
example is in : `tests/k8s_testcases/expected_output/two-way-containment-a-with-different-ipblock-policies-all-pairs.yaml`
The first part of the explanation is the `sub_query_answer_1.output_explanation` which is the result of the first call of `ContainmentQuery.exec()`
and the second part is the  `sub_query_answer_2.output_explanation` which is the result of the second call of `ContainmentQuery.exec()`

Reply:
ok, but wouldn't it be more clear to identify each case based on the `numerical_result `of `query_answer` (between 0 to 3) instead of comparing sub_result to `None`?

Reply:
True, I'll simplify this too